txnbuild/allow_trust.go

```diff
@@ -61,6 +62,20 @@ func (at *AllowTrust) BuildXDR() (xdr.Operation, error) {
 	return op, nil
 }
 
+func assetCodeToCreditAsset(assetCode xdr.AssetCode) (CreditAsset, error) {
+	switch assetCode.Type {
+	case xdr.AssetTypeAssetTypeCreditAlphanum4:
+		code := bytes.Trim(assetCode.AssetCode4[:], "\x00")
+		return CreditAsset{Code: string(code[:])}, nil
+	case xdr.AssetTypeAssetTypeCreditAlphanum12:
+		code := bytes.Trim(assetCode.AssetCode12[:], "\x00")
+		return CreditAsset{Code: string(code[:])}, nil
```

Comment:
🐛  Is the intention to trim `\x00` from both the left and right sides of the byte slice, or only the right? I expect we only need to discard trailing `\x00`s and therefore should use `bytes.TrimRight` because when I've seen padding discussed in documentation (e.g. [SEP-11](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset)) it is only used to pad the tail.
```suggestion
	case xdr.AssetTypeAssetTypeCreditAlphanum4:
		code := bytes.TrimRight(assetCode.AssetCode4[:], "\x00")
		return CreditAsset{Code: string(code[:])}, nil
	case xdr.AssetTypeAssetTypeCreditAlphanum12:
		code := bytes.TrimRight(assetCode.AssetCode12[:], "\x00")
		return CreditAsset{Code: string(code[:])}, nil
```
I realize you're just moving existing code, but I wonder if this is a bug we need to address separately to this PR.

Reply:
Would you mind creating a ticket?

Reply:
(I think it's out of the scope of this PR)

Reply:
👍🏻 https://github.com/stellar/go/issues/3369